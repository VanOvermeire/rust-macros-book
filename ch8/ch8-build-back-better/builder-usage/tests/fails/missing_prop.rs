use builder_macro::Builder;

#[derive(Builder)]
struct Gleipnir {
    roots_of: String,
    breath_of_a_fish: u8,
    anything_else: bool,
}

fn main() {
    // not every property
    Gleipnir::builder()
        .roots_of("mountains".to_string())
        .breath_of_a_fish(1)
        .build();
}
