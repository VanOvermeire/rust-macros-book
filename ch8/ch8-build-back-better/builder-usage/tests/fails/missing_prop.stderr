warning: type `roots_ofOfGleipnirBuilder` should have an upper camel case name
 --> tests/fails/missing_prop.rs:5:5
  |
5 |     roots_of: String,
  |     ^^^^^^^^ help: convert the identifier to upper camel case: `RootsOfOfGleipnirBuilder`
  |
  = note: `#[warn(non_camel_case_types)]` on by default

warning: type `breath_of_a_fishOfGleipnirBuilder` should have an upper camel case name
 --> tests/fails/missing_prop.rs:6:5
  |
6 |     breath_of_a_fish: u8,
  |     ^^^^^^^^^^^^^^^^ help: convert the identifier to upper camel case: `BreathOfAFishOfGleipnirBuilder`

warning: type `anything_elseOfGleipnirBuilder` should have an upper camel case name
 --> tests/fails/missing_prop.rs:7:5
  |
7 |     anything_else: bool,
  |     ^^^^^^^^^^^^^ help: convert the identifier to upper camel case: `AnythingElseOfGleipnirBuilder`

error[E0599]: no method named `build` found for struct `GleipnirBuilder<anything_elseOfGleipnirBuilder>` in the current scope
  --> tests/fails/missing_prop.rs:15:10
   |
3  |   #[derive(Builder)]
   |            ------- method `build` not found for this struct
...
12 |       Gleipnir::builder()
   |       -------------------
   |       |
   |  _____method `build` is available on `GleipnirBuilder<roots_ofOfGleipnirBuilder>`
   | |
13 | |         .roots_of("mountains".to_string())
   | |          --------------------------------- method `build` is available on `GleipnirBuilder<breath_of_a_fishOfGleipnirBuilder>`
14 | |         .breath_of_a_fish(1)
15 | |         .build();
   | |         -^^^^^ method not found in `GleipnirBuilder<anything_elseOfGleipnirBuilder>`
   | |_________|
   |
   |
   = note: the method was found for
           - `GleipnirBuilder<FinalBuilder>`
